sanssouci/lambda_calibration.py

```diff
@@ -50,30 +50,29 @@ def get_perm_p(X, categ, B=100, row_test_fun=stats.ttest_ind):
     """
 
     # Init
-    n = X.shape[0]
-    p = X.shape[1]
+    n, p = X.shape
 
     # Step 1: calculate $p$-values for B permutations of the class assignments
 
     # 1.1: Intialise all vectors and matrices
-    shuffled_categ_current = categ.copy()
+    shuffled_labels = labels.copy()
 
-    shuffled_categ_all = np.zeros([B, n])
+    all_shuffled_labels = np.zeros((B, n))
     for bb in range(B):
-        np.random.shuffle(shuffled_categ_current)
-        shuffled_categ_all[bb, :] = shuffled_categ_current[:]
+        np.random.shuffle(shuffled_labels)
+        all_shuffled_labels[bb] = shuffled_labels
 
     # 1.2: calculate the p-values
     pval0 = np.zeros([B, p])
 
     if row_test_fun == row_welch_tests:  # row Welch Tests (parallelized)
         for bb in range(B):
-            swt = row_welch_tests(X, shuffled_categ_all[bb, :])
-            pval0[bb, :] = swt['p_value'][:]
+            permuted_test_result = row_welch_tests(X, all_shuffled_labels[bb])
+            pval0[bb] = permuted_test_result['p_value']
     else:                     # standard scipy tests
         for bb in range(B):
```

Comment:
```suggestion
        for b in range(B):
``` Not related to your changes, but would be more readable imho